src/mame/drivers/namcos2.cpp

```diff
@@ -5727,8 +5579,8 @@ DRIVER_INIT_MEMBER(namcos2_state,marvland)
 DRIVER_INIT_MEMBER(namcos2_state,metlhawk)
 {
 	/* unscramble sprites */
-	uint8_t *data = memregion("gfx1")->base();
-	for (int i=0; i<0x200000; i+=32*32)
+	uint8_t *data = memregion("sprites")->base();
+	for (int i=0; i<memregion("sprites")->bytes(); i+=32*32)
```

Comment:
Please at least lift the call to `memregion("sprites")->bytes()` out of the loop - right now you're doing a tagmap lookup when it checks the condition on every iteration of the loop.